modules/admin-ui-picard/app/src/reducers/eventDeatilsReducers.js

```diff
@@ -194,6 +198,25 @@ const eventDetails = (state=initialState, action) => {
                 metadata: metadata
             };
         }
+        case SET_EXTENDED_EVENT_METADATA: {
+            const { metadata } = payload;
+
+            const oldExtendedMetadata = state.extendedMetadata;
+            let newExtendedMetadata = [];
+
+            for(const catalog of oldExtendedMetadata){
+                if((catalog.flavor === metadata.flavor) && (catalog.title === metadata.title)){
+                    newExtendedMetadata.push(metadata);
+                } else {
+                    newExtendedMetadata.push(catalog);
+                }
+            }
```

Comment:
put only the state change in the reducer and all preprocessing in a thunk or the method that dispatches the state change